kernel/src/process_standard.rs

```diff
@@ -213,26 +213,27 @@ impl<C: Chip> Process for ProcessStandard<'_, C> {
         self.process_id.get()
     }
 
-    fn enqueue_task(&self, task: Task) -> bool {
+    fn enqueue_task(&self, task: Task) -> Result<(), ErrorCode> {
         // If this app is in a `Fault` state then we shouldn't schedule
         // any work for it.
         if !self.is_active() {
-            return false;
+            return Err(ErrorCode::NODEVICE);
         }
 
-        let ret = self.tasks.map_or(false, |tasks| tasks.enqueue(task));
-
-        // Make a note that we lost this upcall if the enqueue function
-        // fails.
-        if ret == false {
-            self.debug.map(|debug| {
-                debug.dropped_upcall_count += 1;
-            });
-        } else {
-            self.kernel.increment_work();
-        }
-
-        ret
+        self.tasks.map_or_else(
+            |tasks| {
+                tasks.enqueue(task);
+                self.kernel.increment_work();
```

Comment:
This discards the result of `tasks.enqueue` and always increments the work counter, which -- if I understand correctly -- effectively prevent the chip from going to sleep once a task has failed to be enqueued.